core/chains/evm/logpoller/orm.go

```diff
@@ -12,218 +12,205 @@ import (
+	SelectLogsCreatedAfter(ctx context.Context, address common.Address, eventSig common.Hash, after time.Time, confs Confirmations) ([]Log, error)
+	SelectLatestLogByEventSigWithConfs(ctx context.Context, eventSig common.Hash, address common.Address, confs Confirmations) (*Log, error)
+	SelectLatestLogEventSigsAddrsWithConfs(ctx context.Context, fromBlock int64, addresses []common.Address, eventSigs []common.Hash, confs Confirmations) ([]Log, error)
+	SelectLatestBlockByEventSigsAddrsWithConfs(ctx context.Context, fromBlock int64, eventSigs []common.Hash, addresses []common.Address, confs Confirmations) (int64, error)
+	SelectLogsByBlockRange(ctx context.Context, start, end int64) ([]Log, error)
+
+	SelectIndexedLogs(ctx context.Context, address common.Address, eventSig common.Hash, topicIndex int, topicValues []common.Hash, confs Confirmations) ([]Log, error)
+	SelectIndexedLogsByBlockRange(ctx context.Context, start, end int64, address common.Address, eventSig common.Hash, topicIndex int, topicValues []common.Hash) ([]Log, error)
+	SelectIndexedLogsCreatedAfter(ctx context.Context, address common.Address, eventSig common.Hash, topicIndex int, topicValues []common.Hash, after time.Time, confs Confirmations) ([]Log, error)
+	SelectIndexedLogsTopicGreaterThan(ctx context.Context, address common.Address, eventSig common.Hash, topicIndex int, topicValueMin common.Hash, confs Confirmations) ([]Log, error)
+	SelectIndexedLogsTopicRange(ctx context.Context, address common.Address, eventSig common.Hash, topicIndex int, topicValueMin, topicValueMax common.Hash, confs Confirmations) ([]Log, error)
+	SelectIndexedLogsWithSigsExcluding(ctx context.Context, sigA, sigB common.Hash, topicIndex int, address common.Address, startBlock, endBlock int64, confs Confirmations) ([]Log, error)
+	SelectIndexedLogsByTxHash(ctx context.Context, address common.Address, eventSig common.Hash, txHash common.Hash) ([]Log, error)
+	SelectLogsDataWordRange(ctx context.Context, address common.Address, eventSig common.Hash, wordIndex int, wordValueMin, wordValueMax common.Hash, confs Confirmations) ([]Log, error)
+	SelectLogsDataWordGreaterThan(ctx context.Context, address common.Address, eventSig common.Hash, wordIndex int, wordValueMin common.Hash, confs Confirmations) ([]Log, error)
+	SelectLogsDataWordBetween(ctx context.Context, address common.Address, eventSig common.Hash, wordIndexMin int, wordIndexMax int, wordValue common.Hash, confs Confirmations) ([]Log, error)
+}
+
+type orm struct {
 	chainID *big.Int
-	q       pg.Q
+	db      sqlutil.Queryer
 	lggr    logger.Logger
 }
 
-// NewORM creates a DbORM scoped to chainID.
-func NewORM(chainID *big.Int, db *sqlx.DB, lggr logger.Logger, cfg pg.QConfig) *DbORM {
-	namedLogger := logger.Named(lggr, "Configs")
-	q := pg.NewQ(db, namedLogger, cfg)
-	return &DbORM{
+var _ ORM = &orm{}
+
+// NewORM creates an orm scoped to chainID.
+func NewORM(chainID *big.Int, db sqlutil.Queryer, lggr logger.Logger) ORM {
```

Comment:
I saw this on the previous ORMs as well. What's the reason behind returning an Interface instead of the actual struct? AFAIK this is an anti-pattern in go.

Reply:
Wouldn't returning an interface be what we want here? The idea is that when a user wants to use the ORM they don't really care about the implementation, they just want an interface to call the queries with.

Reply:
This is an exported signature, so using an interface locks you in so that methods cannot be added in the future (without breaking the signature). By returning a concrete struct, we are able to expand the method set in a compatible way.

Reply:
https://duncanleung.com/go-idiom-accept-interfaces-return-types/

Reply:
It also makes it easier to test, since unexported methods/fields and even test-only methods can be accessed.

Reply:
In this case don't we need to export orm then since we can't have `NewORM` return an non-exported type? I thought we discussed not exporting this, but I don't mind changing it.

Reply:
Jordan can correct me on this one, but in go it seems ok to return an unexported struct as long as the caller declares the corresponding interface and uses that instead of the struct. Seems aligned with the "accept interfaces return structs" rule.

Reply:
Does our linter issue warnings for this? https://github.com/golang/lint/issues/210

Reply:
Apparently, the pattern is to return the struct **and** export it, which seems very confusing to me since it kinda breaks proper encapsulation. You end up having 2 exported types and only the interface is required.

Reply:
Does anyone have a strong preference here? If not I'll just export the struct and call it a day.